tests/storage/restore.go

```diff
@@ -250,6 +251,25 @@ var _ = SIGDescribe("VirtualMachineRestore Tests", func() {
 		return r
 	}
 
+	createVMWithCloudInit := func(containerDisk cd.ContainerDisk, storageClass string) *v1.VirtualMachine {
+		vm := RenderVMWithRegistryImportDataVolume(containerDisk, storageClass)
+		cloudInitNoCloudSource := &v1.CloudInitNoCloudSource{}
+		cloudInitNoCloudSource.UserDataBase64 = base64.StdEncoding.EncodeToString([]byte(bashHelloScript))
+		vm.Spec.Template.Spec.Domain.Devices.Disks = append(vm.Spec.Template.Spec.Domain.Devices.Disks, v1.Disk{
+			Name: "disk1",
+			DiskDevice: v1.DiskDevice{
+				Disk: &v1.DiskTarget{
+					Bus: v1.DiskBusVirtio,
+				},
+			},
+		})
+		vm.Spec.Template.Spec.Volumes = append(vm.Spec.Template.Spec.Volumes, v1.Volume{
+			Name:         "disk1",
+			VolumeSource: v1.VolumeSource{CloudInitNoCloud: cloudInitNoCloudSource},
+		})
```

Comment:
ACK, refactored this part a bit so we can add additional options to `RenderVMIWithDataVolume`

Reply:
I still think it would be better to totally remove directly access to the vmi spec and instead using the functions from libvmi, but it can be done in a follow-up PR